packages/core/src/content/locales.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Represents the response structure for a query that retrieves a locale.
+ */
+export interface LocaleQueryResponse {
+  locale: Locale;
+}
+
+/**
+ * Represents the response structure for a query that retrieves multiple locales.
+ */
+export interface LocalesQueryResponse {
+  manyLocale: Locale[];
+}
+
+/**
+ * Represents a locale with an id and a label.
+ *
+ * @typedef Locale
+ * @property {string} id - The unique identifier for the locale.
+ * @property {string} label - The display name or label for the locale.
+ */
+export type Locale = {
+  id: string;
+  label: string;
+};
+
+/**
+ * GraphQL query to retrieve a specific locale by its ID.
+ *
+ * @param {string} id - The unique identifier for the locale.
+ * @returns {string} The GraphQL query string.
+ */
+export const GET_LOCALE_QUERY = `
+  query ($id: ID!) {
```

Comment:
I suggest to provide names for the query, like `GetLocaleById` and `GetAllLocales`, it might help with observability, e.g. Cotntent services could gather observability logs on what queries are executed, if they have names - it is easier to filter them.